### PR TITLE
Feature/create resource 39

### PR DESCRIPTION
**Pull request by NeptunFlosse**
The warning 'this.table is undefined' is is caused by the loading component that is displayed at first. 
The application works anyways, so this can be fixed later.

### CODE REVIEW

File: labgrid-web-client/src/app/resource-selector/resource-selector.component.css [hunk 1]
```diff
@@ -0,0 +1,87 @@
+
```
Comment (Simon-Deuring) [inline labgrid-web-client/src/app/resource-selector/resource-selector.component.css:1]: The empty lines can be removed.

File: labgrid-web-client/src/app/resource-selector/resource-selector.component.ts [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        // this resource was added
+        addedResources.push(x);
+      }
+    })
+
+    // console.log('--------results--------')
+    // console.log('removed: ', removedResources)
+    // console.log('added: ', addedResources)
+
+    let response = {successful: false, errorMessage: ''};
```
Comment (Simon-Deuring) [inline labgrid-web-client/src/app/resource-selector/resource-selector.component.ts:123]: I think you also need to handle the newly acquired resources by calling this._rs.acquireResource().

File: labgrid-web-client/src/app/_services/resource.service.ts [hunk 3]
```diff
@@ -94,4 +95,109 @@ export class ResourceService {
...
+            return result[0];
+        }
+    }
+
+
+    public async acquireResource(resourceName: string, place: Place): Promise<{ successful: boolean; errorMessage: string }> {
+        // If the session is already set the data can immediately be read.
+        // Otherwise we wait 1 second.
+        if (this.session) {
+            const result = (await this.session.call('localhost.acquire_resource', [place.name, place.exporter, resourceName]));
```
Comment (Simon-Deuring) [inline labgrid-web-client/src/app/_services/resource.service.ts:148]: In rpc.py the remote procedure acquire_resource takes 4 arguments, not 3.

File: labgrid-web-client/src/app/_services/resource.service.ts [hunk 4]
```diff
@@ -94,4 +95,109 @@ export class ResourceService {
...
+                return { successful: true, errorMessage: '' };
+            } else {
+                return { successful: false, errorMessage: result.error.message };
+            }
+        } else {
+            await new Promise((resolve, reject) => {
+                // The 1000 milliseconds is a critical variable. It may be adapted in the future.
+                setTimeout(resolve, 1000);
+            });
+            const result = (await this.session.call('localhost.acquire_resource', [place.name, place.exporter, resourceName]));
```
Comment (Simon-Deuring) [inline labgrid-web-client/src/app/_services/resource.service.ts:162]: Same as for line 148.

File: labgrid-web-client/src/app/_services/resource.service.ts [hunk 5]
```diff
@@ -94,4 +95,109 @@ export class ResourceService {
             return result[0];
         }
     }
+
+    public async deleteResource(resourceName: string, placeName: string): Promise<Resource> {
+        // If the session is already set the data can immediately be read.
+        // Otherwise we wait 1 second.
+        if (this.session) {
+            const result = (await this.session.call('localhost.delete_resource', [, placeName, resourceName])) as Resource[];
```
Comment (Simon-Deuring) [inline labgrid-web-client/src/app/_services/resource.service.ts:103]: The ',' before placeName should be removed.

**Review by Simon-Deuring — CHANGES_REQUESTED** [review]

File: labgrid-web-client/src/app/resource-selector/resource-selector.component.ts (hunk 2 above)
Comment (NeptunFlosse) [reply labgrid-web-client/src/app/resource-selector/resource-selector.component.ts:123]: Yes, but at the moment there is just a place that can release resources. As soon as releasing works I will add the acquire functionality.

**Review by Simon-Deuring — APPROVED** [review]

**Pull request merged**